x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/10_semantic_text_field_mapping.yml

```
@@ -977,3 +977,451 @@ setup:
+      indices.create:
+        index: test-incompatible-index-options
+        body:
+          settings:
+            index:
+              mapping:
+                semantic_text:
+                  use_legacy_format: false
+          mappings:
+            properties:
+              semantic_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+                index_options:
+                  sparse_vector:
+                    ef_construction: 100
+
+  - match: { status: 400 }
+
+  - do:
+      catch: /\[index_options\] field \[pruning_config\] should only be set if \[prune\] is set to true/
+      indices.create:
+        index: test-incompatible-index-options
+        body:
+          settings:
+            index:
+              mapping:
+                semantic_text:
+                  use_legacy_format: false
+          mappings:
+            properties:
+              semantic_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+                index_options:
+                  sparse_vector:
+                    prune: false
+                    pruning_config:
+                      tokens_freq_ratio_threshold: 18.0
+                      tokens_weight_threshold: 0.6
+
+  - match: { status: 400 }
+
+  - do:
+      catch: /\[tokens_freq_ratio_threshold\] must be between \[1\] and \[100\]/
+      indices.create:
+        index: test-incompatible-index-options
+        body:
+          settings:
+            index:
+              mapping:
+                semantic_text:
+                  use_legacy_format: false
+          mappings:
+            properties:
+              semantic_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+                index_options:
+                  sparse_vector:
+                    prune: true
+                    pruning_config:
+                      tokens_freq_ratio_threshold: 101
+                      tokens_weight_threshold: 0.6
+
+  - match: { status: 400 }
+
+  - do:
+      catch: /unknown field \[some_other_param\]/
+      indices.create:
+        index: test-incompatible-index-options
+        body:
+          settings:
+            index:
+              mapping:
+                semantic_text:
+                  use_legacy_format: false
+          mappings:
+            properties:
+              semantic_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
+                index_options:
+                  sparse_vector:
+                    prune: true
+                    pruning_config:
+                      tokens_freq_ratio_threshold: 18.0
+                      tokens_weight_threshold: 0.6
+                      some_other_param: true
+
+  - match: { status: 400 }
+
+---
+"Specifying sparse vector index options should fail using dense index options":
```

Nitpick: 
```suggestion
"Specifying dense vector index options should fail for sparse embedding models":
```

Can we test the opposite too? Setting sparse vector index options on a dense vector inference ID?